### PR TITLE
chore(javax2jakarta): update transaction api to jakarta

### PR DESCRIPTION
**Pull request by educhastenier**
Update jakarta.transaction:jakarta.transaction-api from 1.3.1 (packages javax.) to 2.0.1 (packages jakarta.)
This requires to update also:
* narayana to a version using jakarta.transaction-api (for tests only)
* geronimo to a version using jakarta.transaction-api (for tests only)

Those packages also needed to update the minimum java version used to 11

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -239,19 +239,15 @@
...
       <scope>test</scope>
       <exclusions>
         <exclusion>
           <artifactId>org.junit.jupiter</artifactId>
           <groupId>junit-jupiter</groupId>
         </exclusion>
-        <exclusion>
-          <artifactId>commons-logging</artifactId>
-          <groupId>commons-logging</groupId>
-        </exclusion>
```
Comment (educhastenier) [inline pom.xml:254]: This module is not a dependency of geronimo-transaction anymore

File: pom.xml [hunk 2]
```diff
@@ -319,7 +315,7 @@
     <commons.jira.pid>12310469</commons.jira.pid>
     <commons.pool.version>2.12.0</commons.pool.version>
     <!-- See DBCP-445 and DBCP-454 -->
-    <commons.osgi.import>javax.transaction;version="1.1.0",javax.transaction.xa;version="1.1.0";partial=true;mandatory:=partial,*</commons.osgi.import>
+    <commons.osgi.import>jakarta.transaction;version="2.0.1",javax.transaction.xa;version="1.1.0";partial=true;mandatory:=partial,*</commons.osgi.import>
```
Comment (educhastenier) [inline pom.xml:318]: I am not sure how this osgi stuff works, so maybe there is something more to change here...

**Comment by garydgregory** [comment]
-1 this will break binary compatibility. 

This type of worked is planned but will be done very differently: we will split the project into a multi-module project to allow support for both javax and Jakarta. But we might complete the pool2 to 3 work first.

**Comment by educhastenier** [comment]
> -1 this will break binary compatibility.
> 
> This type of worked is planned but will be done very differently: we will split the project into a multi-module project to allow support for both javax and Jakarta. But we might complete the pool2 to 3 work first.

Ok, I understand.
I definitely should have read the discussion before starting coding 😉 .
Thx for the feedbacks.

Closing this PR, then.

**Pull request closed without merging**